client/src/main/java/org/apache/falcon/client/FalconClient.java

```diff
@@ -1155,7 +1163,8 @@ private JSONObject getExtensionJobDetailJson(String jobName) {
                 entities = ExtensionHandler.loadAndPrepare(extensionName, jobName, configStream,
                         extensionBuildLocation);
             } catch (Exception e) {
-                throw new FalconCLIException("Error in building the extension", e);
+                LOG.error("Error in building the extension. Cause: ", e);
```

Comment:
May be you should also log this in OUT stream too as the error might come from the extension builder and the error stack will be easily available to the user.